internal/ethapi/simulate.go

```diff
@@ -73,6 +73,19 @@ func (r *simCallResult) MarshalJSON() ([]byte, error) {
 	return json.Marshal((*callResultAlias)(r))
 }
 
+// simBlockResult is the result of a simulated block.
+type simBlockResult struct {
+	sim   *simulator
+	Block *types.Block
```

Comment:
I'm wondering if storing the block here will increase the memory usage of eth simulate over a longer period of blocks

Reply:
Why should this be different? The encoded blocks were previously stored in the map[string]interface{}.